### PR TITLE
New ComponentStateService to manage the components state stored in localStorage

### PR DESCRIPTION
**Pull request by gonzalo-martinez-imt**
Solves #591

### CODE REVIEW

File: projects/ontimize-web-ngx/src/lib/components/list/o-list.component.ts [hunk 1]
```diff
@@ -428,4 +428,11 @@ export class OListComponent extends AbstractOServiceComponent<OListComponentStat
     ObservableWrapper.callEmit(this.onDataLoaded, this.dataResponseArray);
   }
 
+  public registerQuickFilter(arg: any): void {
+    super.registerQuickFilter(arg);
+    if (Util.isDefined(this.quickFilterComponent) && Util.isDefined(this.state.quickFilterActiveColumns)) {
```
Comment (daniel-grana) [inline projects/ontimize-web-ngx/src/lib/components/list/o-list.component.ts:433]: Can we ensure that _this.state_ at this point is initialized? Maybe it should be better to add a checking:
```
... && Util.isDefined(this.state) &&
```

**Review by daniel-grana — CHANGES_REQUESTED** [review]

File: projects/ontimize-web-ngx/src/lib/components/table/extensions/sort/o-mat-sort.ts [hunk 2]
```diff
@@ -1,5 +1,5 @@
-import { Directive, EventEmitter, Output } from '@angular/core';
-import { MatSort, MatSortable } from '@angular/material';
+import { Directive, ElementRef, EventEmitter, Output, Renderer2 } from '@angular/core';
```
Comment (daniel-grana) [inline projects/ontimize-web-ngx/src/lib/components/table/extensions/sort/o-mat-sort.ts:1]: Is correct the import of _Renderer2_? Is it being used anywhere?

**Review by daniel-grana — CHANGES_REQUESTED** [review]

File: projects/ontimize-web-ngx/src/lib/components/table/extensions/sort/o-mat-sort.ts (hunk 2 above)
Comment (gonzalo-martinez-imt) [reply projects/ontimize-web-ngx/src/lib/components/table/extensions/sort/o-mat-sort.ts:1]: It was an unused import

**Review by daniel-grana — APPROVED** [review]

**Review by daniel-grana — DISMISSED** [review]

File: projects/ontimize-web-ngx/src/lib/services/factories.ts [hunk 3]
```diff
@@ -48,6 +49,11 @@ export const O_PERMISSION_SERVICE = new InjectionToken<IPermissionsService>('Per
  */
 export const O_AUTH_SERVICE = new InjectionToken<AuthService>('Authentication service');
 
+// /**
+// * Injection token that can be used to replace the component state service `DefaultComponentStateService`.
+// */
```
Comment (manuel-cid) [inline projects/ontimize-web-ngx/src/lib/services/factories.ts:54]: Remove line comments

**Review by manuel-cid — DISMISSED** [review]

**Review by daniel-grana — APPROVED** [review]

**Review by pacogomezimatia — APPROVED** [review]

**Pull request merged**